tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -0,0 +1,657 @@
+
+    Args:
+        sts_client: The STS client object to be used
+        project_id: The id of the project
+
+    Returns:
+        The STS service account email as a string
+    """
+
+    result = sts_client.googleServiceAccounts().get(
+        projectId=project_id).execute(num_retries=5)
+    return result['accountEmail']
+
+
+def _assign_sts_iam_roles(sts_email, storage_client, project_name, bucket_name,
+                          assign_viewer):
+    """Assign roles to the STS service account that will be required to interact with the bucket.
+
+    Args:
+        sts_email: The email address for the STS service account
+        storage_client: The storage client object used to access GCS
+        project_name: The name of the project
+        bucket_name: The name of the bucket
+        assign_viewer: True if we should also assign the Object Viewer/LegacyReader roles
+    """
+
+    account = 'serviceAccount:' + sts_email
+    bucket = storage_client.bucket(bucket_name, project_name)
+    policy = bucket.get_iam_policy()
+    policy['roles/storage.legacyBucketWriter'].add(account)
+    if assign_viewer:
+        policy[iam.STORAGE_OBJECT_VIEWER_ROLE].add(account)
+        policy['roles/storage.legacyBucketReader'].add(account)
+
+    bucket.set_iam_policy(policy)
+
+
+def _remove_sts_iam_roles(sts_email, storage_client, bucket_name):
+    """Remove the roles that were assigned for the STS service account.
+
+    Args:
+        sts_email: The email address for the STS service account
+        storage_client: The storage client object used to access GCS
+        bucket_name: The name of the bucket
+    """
+
+    account = 'serviceAccount:' + sts_email
+    bucket = storage_client.bucket(bucket_name)
+    policy = bucket.get_iam_policy()
+    policy['roles/storage.legacyBucketWriter'].discard(account)
+    bucket.set_iam_policy(policy)
+
+
+def _add_target_project_to_kms_key(spinner, config, kms_key_name):
+    """Gives the service_account_email the Encrypter/Decrypter role for the given KMS key.
+
+    Args:
+        spinner: The spinner displayed in the console
+        config: A Configuration object with all of the config values needed for the script to run
+        kms_key_name: The name of the KMS key that the project should be given access to
+    """
+
+    kms_client = discovery.build(
+        'cloudkms', 'v1', credentials=config.source_project_credentials)
+
+    # Get the current IAM policy and add the new member to it.
+    crypto_keys = kms_client.projects().locations().keyRings().cryptoKeys()  # pylint: disable=no-member
+    policy_request = crypto_keys.getIamPolicy(resource=kms_key_name)
+    policy_response = policy_request.execute(num_retries=5)
+    bindings = []
+    if 'bindings' in policy_response.keys():
+        bindings = policy_response['bindings']
+    members = []
+    members.append('serviceAccount:' + config.target_service_account_email)
+    new_binding = dict()
```

Comment:
This could also be:

```
    bindings.append({
      'role': 'roles/cloudkms.cryptoKeyEncrypterDecrypter',
      'members': members,
    })
```